src/stories/components/UploadPhoto/UploadPhoto.js

```diff
@@ -142,18 +150,45 @@ export const UploadPhoto = () => {
     }
   };
 
+  const observeFacialDetectionStatus = async (userId, experimentId) => {
+    firestore
+      .collection(firestoreCollections.USER)
+      .doc(userId)
+      .collection(firestoreCollections.EXPERIMENT)
+      .doc(experimentId)
+      .onSnapshot(async (doc) => {
+        const userDoc = doc.data();
+        // TODO: should check if userDoc exists here
+        const facialDetectionStatus = userDoc['facial_detection_status'];
+        if (facialDetectionStatus === 'completed') {
+          setImageFeedback('Photo requirements passed!');
+          setIsImageSatisfied(true);
+        } else {
+          setImageFeedback(facialDetectionStatus);
+          setIsImageSatisfied(false);
+        }
+      });
+    setLoading(false);
```

Comment:
I believe in JS this line will immediately execute without waiting for the update.
So if there is no update, the loading gonna be false anyway. It would be better to put it inside the onSnapshot()

Reply:
ah nice catch! Thank you!